src/number_theory/number_field/basic.lean

```diff
@@ -127,11 +130,81 @@ begin
     (((is_integral_closure.is_integral_algebra ℤ K).is_field_iff_is_field h_inj).mpr hf)
 end
 
-instance [number_field K] : is_dedekind_domain (𝓞 K) :=
+instance : is_dedekind_domain (𝓞 K) :=
 is_integral_closure.is_dedekind_domain ℤ ℚ K _
 
+variables (R : Type*) [comm_ring R] [algebra R K]
+
+lemma _root_.number_field.is_integral_closure.free (hc : is_integral_closure R ℤ K) :
+  free ℤ R :=
+begin
+  haveI : no_zero_smul_divisors ℤ R := function.injective.no_zero_smul_divisors _
+    hc.algebra_map_injective (map_zero _) (map_zsmul _),
+  haveI : is_noetherian ℤ R := is_integral_closure.is_noetherian _ ℚ K _,
+  have basis : Σ n, basis (fin n) ℤ R := free_of_finite_type_torsion_free',
+  obtain ⟨n, b⟩ := basis,
+  exact free.of_basis b,
+end
+
+lemma _root_.number_field.is_integral_closure.is_localization
+  (hc : is_integral_closure R ℤ K) :
+  is_localization (algebra.algebra_map_submonoid R (non_zero_divisors ℤ)) K :=
+begin
+  haveI : char_zero R := char_zero.of_module _ K,
+  haveI : is_domain R := ring_equiv.is_domain (𝓞 K) (number_field.ring_of_integers.equiv R).symm,
```

Comment:
Library-wise it would be nicer to do this by lifting [`integral_closure.is_domain`](https://leanprover-community.github.io/mathlib_docs/ring_theory/integral_closure.html#integral_closure.is_domain) to `is_integral_closure`.